File: algorithms/bipartite_check/cpp/bipartite.cpp
#include<bits/stdc++.h>
using namespace std;
int col[100];
int G[100][100];
int nodes,edges;
bool bipartite(int src)
{
col[src]=1;
queue<int> q;
q.push(src);
while(!q.empty())
{
	int u=q.front();
	q.pop();
	for(int i=1;i<=nodes;i++)
	{
		if(G[u][i]==1 && col[i]==-1)
		{
			col[i]=1-col[u];
			q.push(i);
		}
		else
			if(G[u][i]==1 && col[i]==col[u])
				return false;
	}
}
return true;

}
int main()
{

	cin>>nodes>>edges;
	for(int i=0;i<100;i++)
	{
		col[i]=-1;
	for(int j=0;j<100;j++)
	{
		G[i][j]=0;
	}
}
	for(int i=0;i<edges;i++)
	{
		int u,v;
		cin>>u>>v;
		G[u][v]=1;
		G[v][u]=1;

	}
	cout<<bipartite(1);

}

File: algorithms/bipartite_check/java/Bipartite.java
import java.util.*;
// import java.util.stream.*;

public class Bipartite {
  public static final int n = 100;
  public static final int[] col;
  public static final int[][] G;

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int nodes = sc.nextInt();
    int edges = sc.nextInt();
    init();
    IntStream.range(0, edges).forEach(i -> {
        int u = sc.nextInt();
        int v = sc.nextInt();
        G[u][v] = -1;
        G[u][v] = -1;
    });
  }

  public boolean isBipartite(int start) {
      col[start] = 1;
      Queue queue = new LinkedList();
      queue.push(start);
      while(!queue.isEmpty()) {
        int u = queue.poll();
        IntStream.range(1, nodes + 1).forEach(i -> {
          if (G[u][i] == 1 && col[i] == -1) {
            col[i] = 1 - col[u];
            q.push(i);
          } else {
            if (G[u][i] == 1 && col[i] == col[u]) return false;
          }
        });
      }
  }

  public static void init() {
    col = new int[n];
    G = new int[n][n];
    Arrays.fill(col, -1);
    for(int[] row : G) {
      Arrays.fill(row, 0);
    }
  }
}
